### PR TITLE
Allow to patch headers in testing.Client

### PR DESCRIPTION
**Pull request by yohanboniface**
Ooooold commit that I've been re-implementing today, so maybe it's time to have it upstream ? :)

Useful for example to have an authenticated client:

```
@pytest.fixture
def auth_client(client):
    client.headers['Authorization'] = "Basic 1234"
    return client
```

### CODE REVIEW

File: roll/testing.py [hunk 1]
```diff
@@ -81,6 +81,8 @@ class Client:
     # Default content type for request body encoding, change it to your own
     # taste if needed.
     content_type = 'application/json; charset=utf-8'
+    # Default headers to use eg. for patching Auth in tests.
+    headers = {}
```
Comment (davidbgk) [inline roll/testing.py:85]: Maybe `default_headers` would be less confusing?

File: roll/testing.py [hunk 2]
```diff
@@ -130,6 +132,8 @@ def encode_body(self, body, headers):
     async def request(self, path, method='GET', body=b'', headers=None,
                       content_type=None):
         headers = headers or {}
+        for key, value in self.headers.items():
+            headers.setdefault(key, value)
```
Comment (davidbgk) [inline roll/testing.py:136]: Why not `headers.update(self.default_headers)`?

**Comment by davidbgk** [comment]
Ah, mmh, reading the example in the description would have helped me 😛

File: roll/testing.py (hunk 2 above)
Comment (yohanboniface) [reply roll/testing.py:136]: So `headers` from the current request context (passed as argument) will have precedence over the default ones.

File: roll/testing.py (hunk 1 above)
Comment (yohanboniface) [reply roll/testing.py]: Done :)

**Pull request merged**